apis/applicationintegration/v1alpha1/sqs_types.go

```diff
@@ -0,0 +1,169 @@
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// Enum values for Queue attribute names
+const (
+	AttributeAll                                   string = "All"
+	AttributePolicy                                string = "Policy"
+	AttributeVisibilityTimeout                     string = "VisibilityTimeout"
+	AttributeMaximumMessageSize                    string = "MaximumMessageSize"
+	AttributeMessageRetentionPeriod                string = "MessageRetentionPeriod"
+	AttributeApproximateNumberOfMessages           string = "ApproximateNumberOfMessages"
+	AttributeApproximateNumberOfMessagesNotVisible string = "ApproximateNumberOfMessagesNotVisible"
+	AttributeCreatedTimestamp                      string = "CreatedTimestamp"
+	AttributeLastModifiedTimestamp                 string = "LastModifiedTimestamp"
+	AttributeQueueArn                              string = "QueueArn"
+	AttributeApproximateNumberOfMessagesDelayed    string = "ApproximateNumberOfMessagesDelayed"
+	AttributeDelaySeconds                          string = "DelaySeconds"
+	AttributeReceiveMessageWaitTimeSeconds         string = "ReceiveMessageWaitTimeSeconds"
+	AttributeRedrivePolicy                         string = "RedrivePolicy"
+	AttributeDeadLetterQueueARN                    string = "DeadLetterQueueARN"
+	AttributeMaxReceiveCount                       string = "MaxReceiveCount"
+	AttributeFifoQueue                             string = "FifoQueue"
+	AttributeContentBasedDeduplication             string = "ContentBasedDeduplication"
+	AttributeKmsMasterKeyID                        string = "KmsMasterKeyId"
+	AttributeKmsDataKeyReusePeriodSeconds          string = "KmsDataKeyReusePeriodSeconds"
+)
+
+// Tag is a key value pairs attached to a Amazon SQS queue.
+type Tag struct {
+
+	// The key name that can be used to look up or retrieve the associated value.
+	Key string `json:"key"`
+
+	// The value associated with a key in a tag.
+	Value string `json:"value,omitempty"`
+}
+
+// RedrivePolicy includes the parameters for the dead-letter queue functionality of the source queue.
+type RedrivePolicy struct {
+	// The Amazon Resource Name (ARN) of the dead-letter queue to which Amazon
+	// SQS moves messages after the value of maxReceiveCount is exceeded.
+	// +optional
+	DeadLetterQueueARN *string `json:"deadLetterQueueARN,omitempty"`
+
+	// The number of times a message is delivered to the source queue before
+	// being moved to the dead-letter queue.
+	// +optional
+	MaxReceiveCount *int64 `json:"maxReceiveCount,omitempty"`
+}
+
+// QueueParameters define the desired state of an AWS Queue
+type QueueParameters struct {
+	// The length of time, in seconds, for which the delivery
+	// of all messages in the queue is delayed.
+	// +optional
+	DelaySeconds *int64 `json:"delaySeconds,omitempty"`
+
+	// Designates a queue as FIFO.
+	// +immutable
+	// +optional
+	FifoQueue *bool `json:"fifoQueue,omitempty"`
```

Comment:
```suggestion
	FIFOQueue *bool `json:"fifoQueue,omitempty"`
```

We want to try maintain the Go practice of using uppercase acrnonyms. Note this does apply to struct tags, which should always mirror the AWS API.